synology_dsm/synology_dsm.py

```diff
@@ -221,11 +224,29 @@ def _request(
             params["_sid"] = self._session_id
         if self._syno_token:
             params["SynoToken"] = self._syno_token
-        self._debuglog("Request params: " + str(params))
 
-        # Request data
+        if request_method == "GET":
+            self._debuglog("GET request params: " + str(params))
+
         url = self._build_url(api)
-        response = self._execute_request(request_method, url, params, **kwargs)
+
+        # If the request method is POST and includes data move the params
+        # to the request body. Used to support the weird Syno use of POST
+        # to choose what fields to return. See ./api/core/share.py
+        # for an example.
+        if request_method == "POST" and api == "SYNO.Core.Share":

```

Comment:
"SYNO.Core.Share" ==> Use the class API_KEY

Reply:
Done